jujugui/static/gui/src/app/store/env/api.js

```diff
@@ -668,6 +672,27 @@ YUI.add('juju-env-api', function(Y) {
     },
 
     /**
+      Performs the RedirectInfo request.
+
+      @method redirectInfo
+      @param {Function} callback The callback to call with the response.
+    */
+    redirectInfo: function(callback) {
+      this._send_rpc({
+        type: 'Admin',
+        request: 'RedirectInfo',
+        version: ADMIN_FACADE_VERSION
+      }, resp => {
+        const error = resp.response.error;
+        if (error) {
+          callback(error, resp);
```

Comment:
I find that passing the entire response back to the 'client' in error has been very helpful when debugging issues.